### PR TITLE
Add a table of users

### PR DESCRIPTION
**Pull request by huwshimi**
Done:
- Add a table of (fake) users using the shared table component. Fixes: https://github.com/canonical-web-and-design/maas-squad/issues/1350.
- Add vanilla based table components: Fixes: #38.

QA:
- open the users section in settings
- you should see a list of users

### CODE REVIEW

File: ui/src/app/base/components/MainTable/MainTable.js [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+  } else if (sortDirection === "ascending") {
+    newDirection = "descending";
+  } else {
+    sortKey = null;
+  }
+  _setSortKey(sortKey);
+  _setSortDirection(newDirection);
+};
+
+const _generateHeaders = (
```
Comment (squidsoup) [inline ui/src/app/base/components/MainTable/MainTable.js]: Small thing, but as these aren't class methods, and these functions aren't exported, I'm not sure the `_` prefix is meaningful now? Not overly important though.

File: ui/src/app/settings/containers/Users/Users.js [hunk 2]
```diff
@@ -1,5 +1,122 @@
...
+
+const mapStateToProps = (state, props) => {
+  return {
+    authUser: baseSelectors.auth.getAuthUser(state, props)
+  };
+};
+
+export default connect(
+  mapStateToProps,
+  {}
```
Comment (squidsoup) [inline ui/src/app/settings/containers/Users/Users.js]: I'm not entirely sure what the consequences of passing an empty object here is, but the docs recommend `null`, or `undefined` (we typically use `null`) https://react-redux.js.org/api/connect#mapstatetoprops-state-ownprops-object

File: ui/src/testing/utils.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { shallow } from "enzyme";
+import React from "react";
+
+/**
+ * Assert that some JSX from Enzyme is equal to some provided JSX.
+ * @param {Object} actual - Some JSX from Enzyme.
+ * @param {Object} expected - Some JSX provided in the test.
+ */
+export const compareJSX = (actual, expected) => {
```
Comment (squidsoup) [inline ui/src/testing/utils.js]: Nice!

File: ui/src/app/base/components/MainTable/MainTable.js [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+        _generateHeaders(
+          currentSortKey,
+          currentSortDirection,
+          expanding,
+          headers,
+          sortable,
+          _setSortKey,
+          _setSortDirection
+        )}
+      {headers &&
```
Comment (squidsoup) [inline ui/src/app/base/components/MainTable/MainTable.js]: as discussed on irc, this should probably be `rows`

File: ui/src/app/base/components/MainTable/MainTable.js [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+  // account for the extra cell in the body rows.
+  return (
+    <thead>
+      <TableRow>
+        {headerItems}
+        {expanding && <TableHeader className="u-hide"></TableHeader>}
+      </TableRow>
+    </thead>
+  );
+};
```
Comment (squidsoup) [inline ui/src/app/base/components/MainTable/MainTable.js]: Tiny nit, but nice to have a line between functions.

File: ui/src/app/base/components/MainTable/MainTable.js (hunk 1 above)
Comment (huwshimi) [reply ui/src/app/base/components/MainTable/MainTable.js]: Done.

File: ui/src/app/base/components/MainTable/MainTable.js (hunk 5 above)
Comment (huwshimi) [reply ui/src/app/base/components/MainTable/MainTable.js]: Oops, thanks.

File: ui/src/app/base/components/MainTable/MainTable.js (hunk 4 above)
Comment (huwshimi) [reply ui/src/app/base/components/MainTable/MainTable.js]: Thanks.

File: ui/src/app/settings/containers/Users/Users.js (hunk 2 above)
Comment (huwshimi) [reply ui/src/app/settings/containers/Users/Users.js]: Thanks, done.

**Pull request merged**